Paltineanu Rares-Mihai/curs10 - 3.4/main.c

```diff
@@ -0,0 +1,326 @@
+/**Write a program to compute the value of a polynomial of degree n with complex coefficients for a given complex value.
+ Use functions o implement computations
+*/
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+
+typedef struct {
+
+    char *coeff;
+} polynomial;
+
+void initialize(polynomial *v, int n) {
+
+    for(int i = 0; i <= n; i++) {
+
+        v[i].coeff = (char *)malloc(sizeof(char) * 10);
+
+    }
+}
+
+void read(polynomial *v, int n) {
+
+    printf("Enter the coefficients of the polynomial:\n");
+
+    for(int i = 0; i <= n; i++) {
+
+        gets(v[i].coeff);
+
+        while(strchr(v[i].coeff, 'i') == NULL) {
+
+            printf("Please read a complex coefficient.\n");
+            gets(v[i].coeff);
+        }
+
+    }
+}
+
+long combination(int n, int k) {
+
+    long nr = 1;
+
+    if(k == 0) {
+        return 1;
+    }
+
+    if(k == n) {
+        return 1;
+    }
+
+    if(k == 1) {
+        return n;
+    }
+
+    for(int i = 0; i < k; i++) {
+
+        nr = nr * (n - i) / (k - i);
+
+    }
+
+    return nr;
+
+}
+
+void takeTheValues(long *real, long *imaginary, char *vect) {
+
+    ///I know that i could have in the struct int real, imaginary; and then read the coefficients, but i wanted to make the things more complicated
```

Comment:
Don't get used to over-complicating your life, it's not going to get any easier!

Reply:
Okk